File: Jazz.web.frame/java/jazz.common.jms/src/org/jazz/common/jms/jmsService.java
package org.jazz.common.jms;

import java.util.function.Function;

import javax.jms.JMSException;
import javax.jms.Message;
import javax.jms.MessageConsumer;
import javax.jms.MessageProducer;

public class jmsService {
   public static void PTPsend(jmsData data,jmsConfig config) throws JMSException
   {
	  MessageProducer producer= config.getNewProducer();
	  producer.send(data.get_mess());
	  producer.close();
   }
   
   public static jmsResult PTPreceive(jmsConfig config,Function<Message,Object> onMessagefunc) throws JMSException
   {
	  MessageConsumer consumer=config.getNewConsumer();
	  jmsResult res=new jmsResult();
//	  consumer.setMessageListener(new MessageListener() {
//          @Override
//          public void onMessage(Message message) {
//        	 res.setData(onMessagefunc.apply(message));
//          }
//      });
//	  consumer.close();
	  Message receive = consumer.receive();
//	  receiveMess
	   if(null != receive){
	      res.setData(onMessagefunc.apply(receive));
	      // receive.acknowledge();
	   }else{
	       //没有值嘛
		   System.out.println("it not have message");
	   //
	   }
	  
	  return res;
   }
   
   public static jmsResult PTPreceive(MessageConsumer consumer,Function<Message,Object> onMessagefunc) throws JMSException
   {
	  jmsResult res=new jmsResult();
	  Message receive = consumer.receive();
	   if(null != receive){
	      res.setData(onMessagefunc.apply(receive));
	      // receive.acknowledge();
	   }else{
	       //没有值嘛
		   System.out.println("it not have message");
	   //
	   }
	  
	  return res;
   }
   
   public static void TOPsend(jmsData data,jmsConfig config)
   {
   }
   
   public static jmsResult TOPreceive(jmsConfig config)
   {
   
	   return null;
   }
}

File: Jazz.web.frame/net/Jazz.Common.JMS/JmsService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Apache.NMS;
using Apache.NMS.ActiveMQ;
using Apache.NMS.Util;


namespace Jazz.Common.JMS
{
    public static class JmsService
    {
        public static void PTPsend(this IMessageProducer producer, JmsData Data)
        {
            producer.Send(Data.Data);
        }

        public static void PTPsend(JmsConfig Config,JmsData Data) 
        {
            using (var producer = Config.getNewProducer())
            {
                producer.Send(Data.Data);
            }
        }

        public static JmsResult PTPreceive( this IMessageConsumer consumer,  Func<IMessage, Object> onMessagefunc)
        {
            JmsResult res = new JmsResult();
            IMessage receive = consumer.Receive();
            if (null != receive)
            {
                res.Data = onMessagefunc(receive);
                //receive.Acknowledge();
            }
            else
            {
            }
            return res;
        }

        public static JmsResult PTPreceive(JmsConfig Config,Func<IMessage,Object> onMessagefunc)
        {
            using (var consumer = Config.getNewConsumer())
            {
                JmsResult res = new JmsResult();
                IMessage receive = consumer.Receive();
                if (null != receive)
                {
                    res.Data = onMessagefunc(receive);
                    //receive.Acknowledge();
                }
                else
                {
                }
                return res;
            }
        }
   
        public static void TOPsend(JmsData data,JmsConfig config)
        {
        }

        public static JmsResult TOPreceive(JmsConfig config)
        {

            return null;
        }
   
    }
}
